src/systems/CMakeLists.txt

```diff
@@ -33,6 +33,17 @@ function(gz_add_system system_name)
 
   set(system_name ${system_name}-system)
 
+
+  if (MSVC)
+    # Warning #4251 is the "dll-interface" warning that tells you when types used
+    # by a class are not being exported. These generated source files have private
+    # members that don't get exported, so they trigger this warning. However, the
+    # warning is not important since those members do not need to be interfaced
+    # with.
+    set_source_files_properties(${sources} COMPILE_FLAGS "/wd4251 /wd4146")
```

Comment:
I know we suppress 4146 in some other libraries, but it doesn't seem to be related to the dll interface, right?

https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-2-c4146?view=msvc-160

Reply:
uh that is right, I copy paste the chunk of code without looking well into it. What do you prefer, remove it or fix the comment?

Reply:
I'm leaning towards removing and addressing that warning on the code

Reply:
d38c8f3